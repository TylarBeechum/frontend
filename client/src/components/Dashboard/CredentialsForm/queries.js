import client, { gql } from '../../../clientQuery';

const queries = {
  addNewCredentials(body) {
    return client.mutate({
      variables: {
        credName: body.credName,
        ownerName: body.ownerName,
        description: body.description,
        type: body.type,
        studentEmail: body.studentEmail,
        imageUrl: body.imageUrl,
        criteria: body.criteria,
        issuedOn: body.issuedOn,
        expirationDate: body.expirationDate,
        schoolId: body.schoolId
      },
      mutation: gql`
        mutation AddNewCredential(
          $credName: String!
          $ownerName: String!
          $description: String!
          $type: String!
          $studentEmail: String!
          $imageUrl: String!
          $criteria: String!
          $issuedOn: String!
          $expirationDate: String
          $schoolId: ID!
        ) {
          addNewCredential(
            credName: $credName
            ownerName: $ownerName
            description: $description
            type: $type
            studentEmail: $studentEmail
            imageUrl: $imageUrl
            criteria: $criteria
            issuedOn: $issuedOn
            expirationDate: $expirationDate
            schoolId: $schoolId
          ) {
            credName
            ownerName
            txHash
            description
            studentEmail
            imageUrl
            issuedOn
            expirationDate
            schoolId
          }
        }
      `
    });
  }
};

export default queries;
