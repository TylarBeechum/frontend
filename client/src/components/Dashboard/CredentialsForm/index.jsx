import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { MaskedInput, Select, Box, Heading, TextArea } from 'grommet';

import {
  BaseForm,
  BaseTextInput,
  BaseFormField,
  BaseButton
} from '../../../styles/themes';

import queries from './queries';
import Field from '../../Field';

const CredentialsForm = ({ history }) => {
  const [credsInputs, setCredsInputs] = useState({
    name: '',
    description: '',
    type: '',
    studentEmail: '',
    imageUrl: '',
    criteria: '',
    issuedOn: '',
    expirationDate: '',
    schoolId: localStorage.id
  });

  const handleChanges = e => {
    setCredsInputs({
      ...credsInputs,
      [e.target.name]: e.target.value
    });
  };

  const handleSubmit = async e => {
    e.preventDefault();
    try {
      history.push('/dashboard');
      await queries.addNewCredentials(credsInputs);
    } catch (error) {
      console.error(error);
    }
  };

  return (
    <Container>
      <section>
        <h2>Issue Credential</h2>
        <BaseForm onSubmit={handleSubmit}>
          <Box>
            <BaseFormField label="Name of Student">
              <BaseTextInput
                name="ownerName"
                placeholder="Jane Doe"
                onChange={handleChanges}
                value={credsInputs.ownerName}
                required
              />
            </BaseFormField>
            <BaseFormField label="Name of Student">
              <BaseTextInput
                name="credName"
                placeholder="Masters in Philopsophy"
                onChange={handleChanges}
                value={credsInputs.credName}
                required
              />
            </BaseFormField>
            <BaseFormField label="Type">
              <BaseTextInput
                name="type"
                placeholder="Masters, PhD, Cert, etc."
                onChange={handleChanges}
                value={credsInputs.type}
                required
              />
            </BaseFormField>
            <BaseFormField label="Description">
              <TextArea
                name="description"
                placeholder="Summary of credential"
                onChange={handleChanges}
                value={credsInputs.description}
                required
              />
            </BaseFormField>

            <BaseFormField label="Student Email">
              <BaseTextInput
                name="studentEmail"
                placeholder="Jane.Doe@gmail.com"
                onChange={handleChanges}
                value={credsInputs.studentEmail}
                required
              />
            </BaseFormField>
            <BaseFormField label="School Seal Image URL">
              <BaseTextInput
                name="imageUrl"
                placeholder="Image"
                onChange={handleChanges}
                value={credsInputs.imageUrl}
                required
              />
            </BaseFormField>
            <BaseFormField label="Criteria">
              <BaseTextInput
                name="criteria"
                placeholder="Enter the criteria for the credentials"
                onChange={handleChanges}
                value={credsInputs.criteria}
                required
              />
            </BaseFormField>
            <BaseFormField label="Issued Date">
              <BaseTextInput
                name="issuedOn"
                placeholder="Enter the issue date for the credentials"
                onChange={handleChanges}
                value={credsInputs.issuedOn}
                required
              />
            </BaseFormField>
            <BaseFormField label="Expiration Date">
              <BaseTextInput
                name="expirationDate"
                placeholder="Enter the expiration date for the credentials"
                onChange={handleChanges}
                value={credsInputs.expirationDate}
              />
            </BaseFormField>
            <BaseButton
              margin="medium"
              type="submit"
              primary
              label="Submit"
              alignSelf="center"
            />
          </Box>
        </BaseForm>
      </section>
    </Container>
  );
};

CredentialsForm.propTypes = {
  history: PropTypes.object.isRequired // eslint-disable-line react/forbid-prop-types
};

const Container = styled.main`
  width: 100%;
  height: calc(100vh - 70px);
  padding: 120px 3% 0;
  position: relative;

  section {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    width: 375px;
    height: 100vh;
    background: ${props => props.theme.global.colors.dashBoardBg};
    padding: 120px 3% 0;
    border-left: 1px solid ${props => props.theme.global.colors.dashBoardBorder};
    overflow-x: hidden;
    overflow-y: auto;

    h2 {
      width: 100%;
      text-align: center;
      margin-bottom: 37px;
    }
    form {
      width: 100%;
      display: flex;
      flex-direction: column;
    }
  }
`;

export default CredentialsForm;
