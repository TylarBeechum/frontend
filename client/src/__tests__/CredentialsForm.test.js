/* eslint-disable */
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import {
  renderWithRouterAndProviders,
  rendererWithRouterAndProviders
} from '../__testHelpers__';

import CredentialsForm from '../components/Dashboard/CredentialsForm';

describe('<CredentialsForm />', () => {
  describe('should render the CredCard component successfully without crashing', () => {
    it('tests that the component does not crash when it is rendered', () => {
      const helpers = renderWithRouterAndProviders(
        <CredentialsForm history={{ location: { pathname: '/' } }} />
      );
    });
  });

  describe('should match the previous snapshot of the component', () => {
    it('tests that the component has not changed and matched the previous snapshot', () => {
      const tree = rendererWithRouterAndProviders(
        <CredentialsForm history={{ location: { pathname: '/' } }} />
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
