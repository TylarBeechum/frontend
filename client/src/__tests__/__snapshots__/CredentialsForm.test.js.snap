// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CredentialsForm /> should match the previous snapshot of the component tests that the component has not changed and matched the previous snapshot 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 iDwiuh"
>
  <main
    className="sc-bZQynM bYzBMv"
  >
    <div
      className="sc-htoDjs faMgFv"
    >
      <section>
        <div>
          <img
            alt="school seal"
            src="certEmblem.png"
          />
        </div>
        <h1>
          [Credential Name]
        </h1>
        <h3>
          [Description]
        </h3>
        <h3>
          Issued on:
           [August 10, 2019]
        </h3>
        <h3>
          Issued by:
          testName
        </h3>
        <h2>
          John Doe
        </h2>
      </section>
      <div />
    </div>
    <section
      className="sc-gzVnrw feuSWF"
    >
      <h2>
        Issue Credential
      </h2>
      <form
        className="sc-htpNat KtFsv"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          className="StyledBox-sc-13pk1d4-0 IRSNj"
        >
          <div
            className="sc-ifAKCX sc-EHOje fclhkO FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
          >
            <label
              className="StyledText-sc-1sadyjn-0 bKLvLm"
            >
              Name of Student
            </label>
            <div
              className="StyledBox-sc-13pk1d4-0 gZSidW"
            >
              <div
                className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kYQCpm"
              >
                <input
                  autoComplete="off"
                  className="sc-bxivhb bUZGRr StyledTextInput-sc-1x30a0s-0 duvNfq"
                  name="ownerName"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Jane Doe"
                  required={true}
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="sc-ifAKCX sc-EHOje fclhkO FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
          >
            <label
              className="StyledText-sc-1sadyjn-0 bKLvLm"
            >
              Credential Name
            </label>
            <div
              className="StyledBox-sc-13pk1d4-0 gZSidW"
            >
              <div
                className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kYQCpm"
              >
                <input
                  autoComplete="off"
                  className="sc-bxivhb bUZGRr StyledTextInput-sc-1x30a0s-0 duvNfq"
                  name="credName"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Masters in Philosophy"
                  required={true}
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="sc-ifAKCX sc-EHOje fclhkO FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
          >
            <label
              className="StyledText-sc-1sadyjn-0 bKLvLm"
            >
              Type
            </label>
            <div
              className="StyledBox-sc-13pk1d4-0 gZSidW"
            >
              <div
                className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kYQCpm"
              >
                <input
                  autoComplete="off"
                  className="sc-bxivhb bUZGRr StyledTextInput-sc-1x30a0s-0 duvNfq"
                  name="type"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Masters, PhD, Cert, etc."
                  required={true}
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="sc-ifAKCX sc-EHOje fclhkO FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
          >
            <label
              className="StyledText-sc-1sadyjn-0 bKLvLm"
            >
              Description
            </label>
            <div
              className="StyledBox-sc-13pk1d4-0 gZSidW"
            >
              <textarea
                className="StyledTextArea-sc-17i3mwp-0 gIdccB"
                name="description"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="Student demostrated ability..."
                required={true}
                value=""
              />
            </div>
          </div>
          <div
            className="sc-ifAKCX sc-EHOje fclhkO FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
          >
            <label
              className="StyledText-sc-1sadyjn-0 bKLvLm"
            >
              Student Email
            </label>
            <div
              className="StyledBox-sc-13pk1d4-0 gZSidW"
            >
              <div
                className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kYQCpm"
              >
                <input
                  autoComplete="off"
                  className="sc-bxivhb bUZGRr StyledTextInput-sc-1x30a0s-0 duvNfq"
                  name="studentEmail"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Jane.Doe@gmail.com"
                  required={true}
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="sc-ifAKCX sc-EHOje fclhkO FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
          >
            <label
              className="StyledText-sc-1sadyjn-0 bKLvLm"
            >
              School Seal Image URL
            </label>
            <div
              className="StyledBox-sc-13pk1d4-0 gZSidW"
            >
              <div
                className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kYQCpm"
              >
                <input
                  autoComplete="off"
                  className="sc-bxivhb bUZGRr StyledTextInput-sc-1x30a0s-0 duvNfq"
                  name="imageUrl"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="www.image.com/schoolSeal"
                  required={true}
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="sc-ifAKCX sc-EHOje fclhkO FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
          >
            <label
              className="StyledText-sc-1sadyjn-0 bKLvLm"
            >
              Criteria
            </label>
            <div
              className="StyledBox-sc-13pk1d4-0 gZSidW"
            >
              <div
                className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kYQCpm"
              >
                <input
                  autoComplete="off"
                  className="sc-bxivhb bUZGRr StyledTextInput-sc-1x30a0s-0 duvNfq"
                  name="criteria"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Completed studies in..."
                  required={true}
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="sc-ifAKCX sc-EHOje fclhkO FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
          >
            <label
              className="StyledText-sc-1sadyjn-0 bKLvLm"
            >
              Issued Date
            </label>
            <div
              className="StyledBox-sc-13pk1d4-0 gZSidW"
            >
              <div
                className="StyledMaskedInput__StyledMaskedInputContainer-sc-99vkfa-1 kyyRCm"
              >
                <input
                  autoComplete="off"
                  className="sc-dnqmqq bDsesi StyledMaskedInput-sc-99vkfa-0 kedzFm"
                  name="issuedOn"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="M/D/YYYY"
                  required={true}
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="sc-ifAKCX sc-EHOje fclhkO FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
          >
            <label
              className="StyledText-sc-1sadyjn-0 bKLvLm"
            >
              Expiration Date
            </label>
            <div
              className="StyledBox-sc-13pk1d4-0 gZSidW"
            >
              <div
                className="StyledMaskedInput__StyledMaskedInputContainer-sc-99vkfa-1 kyyRCm"
              >
                <input
                  autoComplete="off"
                  className="sc-dnqmqq bDsesi StyledMaskedInput-sc-99vkfa-0 kedzFm"
                  name="expirationDate"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="M/D/YYYY"
                  value=""
                />
              </div>
            </div>
          </div>
          <button
            className="sc-bdVaJa bDWFJH StyledButton-sc-323bzc-0 geFpyw"
            disabled={false}
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </section>
  </main>
</div>
`;
